File: src/main/java/no/nav/pensjon/innsyn/domain/Dagpenger.java
package no.nav.pensjon.innsyn.domain;

public class Dagpenger {

    private final double ferietillegg;
    private final double barnetillegg;
    private final double dagpenger;
    private final double uavkortetGrunnlag;
    private final int arDagpengerUtbetalt;
    private final String kilde;
    private final String status;
    private final String type;
    private final String rapporttype;

    public Dagpenger(double ferietillegg,
                     double barnetillegg,
                     double dagpenger,
                     double uavkortetGrunnlag,
                     int arDagpengerUtbetalt,
                     String kilde,
                     String status,
                     String type,
                     String rapporttype) {
        this.ferietillegg = ferietillegg;
        this.barnetillegg = barnetillegg;
        this.dagpenger = dagpenger;
        this.uavkortetGrunnlag = uavkortetGrunnlag;
        this.arDagpengerUtbetalt = arDagpengerUtbetalt;
        this.kilde = kilde;
        this.status = status;
        this.type = type;
        this.rapporttype = rapporttype;
    }

    public double getFerietillegg() {
        return ferietillegg;
    }

    public double getBarnetillegg() {
        return barnetillegg;
    }

    public double getDagpenger() {
        return dagpenger;
    }

    public double getUavkortetGrunnlag() {
        return uavkortetGrunnlag;
    }

    public int getArDagpengerUtbetalt() {
        return arDagpengerUtbetalt;
    }

    public String getKilde() {
        return kilde;
    }

    public String getStatus() {
        return status;
    }

    public String getType() {
        return type;
    }

    public String getRapporttype() {
        return rapporttype;
    }
}

File: src/main/kotlin/no/nav/pensjon/innsyn/popp/domain/Dagpenger.kt
package no.nav.pensjon.innsyn.popp.domain

import no.nav.pensjon.innsyn.common.domain.Domain
import no.nav.pensjon.innsyn.popp.domain.support.DagpengerStatus
import no.nav.pensjon.innsyn.popp.domain.support.DagpengerType
import no.nav.pensjon.innsyn.popp.domain.support.KildeType
import no.nav.pensjon.innsyn.popp.domain.support.RapportType
import javax.persistence.*

@Entity
@Table(name = "T_DAGPENGER")
data class Dagpenger(
        @Id
        @Column(name = "DAGPENGER_ID")
        val id: Int,
        @Column(name = "PERSON_ID")
        val personId: Int,
        @Column(name = "FERIETILLEGG")
        val ferietillegg: Double?,
        @Column(name = "BARNETILLEGG")
        val barnetillegg: Double?,
        @Column(name = "DAGPENGER")
        val dagpenger: Double,
        @Column(name = "UAVKORTET_DP_GRLAG")
        val uavkortetGrunnlag: Double?,
        @Column(name = "AR")
        val arDagpengerUtbetalt: Int,
        @ManyToOne
        @JoinColumn(name = "K_KILDE_T")
        private val kildeType: KildeType,
        @ManyToOne
        @JoinColumn(name = "K_DAGPENGER_S")
        private val dagpengerStatus: DagpengerStatus,
        @ManyToOne
        @JoinColumn(name = "K_DAGPENGER_T")
        private val dagpengerType: DagpengerType,
        @ManyToOne
        @JoinColumn(name = "K_RAPPORT_T")
        private val rapportType: RapportType

) : Domain {
    @get:Transient
    val kilde
        get() = kildeType.dekode

    @get:Transient
    val status
        get() = dagpengerStatus.dekode

    @get:Transient
    val type
        get() = dagpengerType.dekode

    @get:Transient
    val rapporttype
        get() = rapportType.dekode

    @get:Transient
    override val fields
        get() = setOf(::ferietillegg, ::barnetillegg, ::dagpenger, ::uavkortetGrunnlag, ::arDagpengerUtbetalt,
                ::kilde, ::status, ::type, ::rapporttype)
}